sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigtable/BigtableIO.java

```diff
@@ -382,6 +399,14 @@ BigtableService getBigtableService(PipelineOptions pipelineOptions) {
         return bigtableService;
       }
       BigtableOptions.Builder clonedOptions = options.toBuilder();
+      // Override the projectId in the options with the runtime-provided  projectId.
+      if(projectId != null) {
+        clonedOptions.setProjectId(projectId.get());
+      }
+      // Override the instanceId in the options with the runtime-provided  projectId.
+      if(instanceId != null) {
+        clonedOptions.setInstanceId(instanceId.get());
```

Comment:
Should make it clear in the Javadoc that the *only* part of the BigtableOptions that are overridable via parameters are the project and instance ID, etc.